Java-Loops.md

```diff
@@ -0,0 +1,208 @@
+
+### The While loop
+
+The `while` statement evaluates the `expression` (which must return a boolean value), and if it is `true`, the `Statements` get executed.
+
+```java
+while (expression) 
+{
+  // Statements
+}
+```
+
+In the following example, the `expression` is given by `iter_While < 10`. As we increment `iter_While` by `1` each time the loop is executed, the `while` loop will keep going till `iter_While` reaches `10`.
+
+```java
+int iter_While = 0;
+while (iter_While < 10) 
+{
+    System.out.print (iter_While + " ");
+    // Increment the counter
+    // Iterated 10 times, iter_While 0,1,2...9
+    iter_While++;
+}
+System.out.println ("iter_While Value: " + iter_While);
+```
+
+Output:
+
+    0 1 2 3 4 5 6 7 8 9
+    iter_While Value: 10
+
+### The Do While loop
+
+The `do while` is very similar to the `while` loop in the way it works, but is *exit controlled* (unlike the `for` and `while` loops which are *entry controlled*), that is, the truth value of its `expression` is evaluated after the execution of `Statements`.
+
+```java
+do 
+{
+    // Statements
+} 
+while (expression);
+```
+
+This kind of loop is particularly useful if you want your `Statements` to be executed atleast once, irrespective of what `expression` evaluates to. You want to do this if you are initializing a variable inside your loop and plan on using its value later.
+
+```java
+int iter_DoWhile = 20;
+do 
+{
+    System.out.print (iter_DoWhile + " ");
+
+    // Increment the counter
+    iter_DoWhile++;
+} 
+while(iter_DoWhile < 10);
+System.out.println ("iter_DoWhile Value: " + iter_DoWhile);
+```
+
+Output:
+
+    20
+    iter_DoWhile Value: 21
+
+### The For loop
+
+The `for` loop give you a compact way to iterate over a range of values.
+
+```java
+for (initialization; expression; increment) 
+{
+    // Statements
+}
+```
+
+* `initialization` - Initializes the loop and is executed just once, at the beginning.
+* `expression` - Evaluated at the beginning of each iteration. If the `expression` evaluates to `true`, `Statements` will get executed.
+* `increment` - Invoked after each iteration through the loop. You can increase/decrease the value of variables here.
+
+```java
+for (int iter_For = 0; iter_For < 10; iter_For++) 
+{
+    System.out.print (iter_For + " ");
+    // Iterated 10 times, iter_For 0,1,2...9
+}
+System.out.println("iter_For Value: " + fooFor);
+```
+
+Output:
+
+    0 1 2 3 4 5 6 7 8 9
+    iter_For Value: 10
+
+### The For Each loop
+
+Also called the Enhanced For loop, it is an extremely useful and simple way to iterate over each item in a collection, arrays as well as objects that implement the Iterable interface.
+
+```java
+for (object : iterable)
+{
+    // Statements
+}
+```
+
+The loop is read as for each element in the `iterable` (could be an array, collectable etc.). The `object` type must match the element type of the `iterable`. 
+
+```java
+int[] number_list = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
```

Comment:
Array is not yet covered. So, clearly mention that if they don't know about Array, they don't need to worry about it. Then give a one line description of what Array is.